components/ArticleList.jsx

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from "react";
+import { fetchAllArticles } from "../utils/utils";
+import ArticleCard from "./ArticleCard";
+
+const ArticleList = () => {
+  const [articleList, setArticleList] = useState([])
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    fetchAllArticles().then(({ articles }) => {
```

Comment:
loading should be false by default, true when waiting and then false again once the data has come back